### PR TITLE
Dataset string memcap 3910/v15

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/3910

Previous PR: https://github.com/OISF/suricata/pull/11296

Changes since v14:
- remove THashData size check when checking for memcap for a value dequeued from the spare queue
- rebased on top of latest master

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1944

### CODE REVIEW

**Comment by victorjulien** [comment]
Passed my QA. Ran this PR with SV PR https://github.com/OISF/suricata-verify/pull/1944. Local pipeline 4962, run 506.

**Review by victorjulien — APPROVED** [review]

**Comment by victorjulien** [comment]
Will need a rebase after #11394 is merged, got a conflict when trying to stage them together.

**Comment by inashivb** [comment]
> Will need a rebase after #11394 is merged, got a conflict when trying to stage them together.

JFYI, the conflict is on [5d46096](https://github.com/OISF/suricata/pull/11384/commits/5d46096e96df716fbd4321ec40603dde068f8402) so I'll wait for the merge of your PR.

**Pull request closed without merging**
